### ISSUE TITLE
Setup URL aliases for functions, etc.

### ISSUE THREAD

**Issue by quicksketch** [backdrop-ops/docs.backdropcms.org#14]
Right now the URLs on the API site are either `node/[nid]` or `api/backdrop/core!includes!actions.inc/function/actions_execute/1`.

These long URLs for API docs are often necessary when the same function or constant is defined multiple times (e.g. BACKDROP_ROOT is in `index.php`, `update.php`, `install.php`, etc.), but we have a ton of things that are only defined once: functions, classes, etc. I think these should probably be aliased to make them easier to find. I've always appreciated that php.net is just http://php.net/strpos for example.

Proposed:
- `/function_name` for all functions.
- `/ClassName` for all classes.
- Files and constants are left as-is.
- If a function or class is defined twice, there is no alias, but the 404 page should do a search.
- The alias should point at the latest recommended release version (e.g. 1.x or 2.x) once we have multiple versions.

I don't know what we should do about handbook pages, especially important ones like the Form API Reference (https://api.backdropcms.org/node/41642). It seems like they should be shortened to permanent URLs as well, e.g. https://api.backdropcms.org/formapi

**Comment by klonos** [comment]
Yeah, long since this issue has been filed and I have to admit that I don't hang around the API site, but when I do visit it (mainly in order to review a handbook page etc.), it always bothers me that we still have some URLs with short, meaningful paths (like [https://api.backdropcms.org/glossary](https://api.backdropcms.org/glossary) for example in the front page) while most of the rest are ```/node/12345```.

Can we please do a "quick and dirty" thing and turn pathauto on for handbook pages (the [Developer Documentation](https://api.backdropcms.org/documentation) part) and set it to something like ```[node:menu-link:parent:url]/[node:title]```

It will be an SEO+ too.

PS: Does it make sense to separate the doc pages from the API site? ...or are these API docs?

**Comment by quicksketch** [comment]
> Can we please do a "quick and dirty" thing and turn pathauto on for handbook pages (the Developer Documentation part) and set it to something like [node:menu-link:parent:url]/[node:title]

Sure, we updated the API site to the latest version not too long ago, so Pathauto is there and available. It looks like our current handbook pages are actually just `[node:title]`-like patterns, which is nice and short and allows us to rearrange the handbook in the future without changing the URLs or setting up redirects. I've enabled this on the current site and generated new URLs.

> PS: Does it make sense to separate the doc pages from the API site? ...or are these API docs?

We decided to keep them all together for the time being. We actually even moved docs (e.g. coding standards) from the main site over to the API site, because it fit better there.

We still need to make aliases for functions/classes, as that setup is a bit more complicated.

**Comment by klonos** [comment]
It's looking great! Thanx.

I still think that ```[node:menu-link:parent:url]/[node:title]``` would be better to signify structure in the URL and add to the SEO, but then again it would create long URLs which is not nice. Besides, who looks at the URL for structure when there's a breadcrumb? ...besides people with self-diagnosed OCD like myself :smile:

**Comment by jenlampton** [comment]
Sounds like this is done? Please re-open if that's not the case.

**Comment by BWPanda** [comment]
This hasn't been done yet. I just came to this issue queue ready to open an issue about doing just this (creating short aliases for function names), only to find this existing issue from years ago.

So the main purpose of this issue hasn't been addressed yet. For those of you who like this sort of thing, here's one of those use-case-thingys...

As a busy (lazy?) developer, I want to type `api.backdropcms.org/backdrop_get_messages` into my browser's URL bar and be automatically redirected to `https://api.backdropcms.org/api/backdrop/core%21includes%21bootstrap.inc/function/backdrop_get_messages/1`.
